grafana-operator/grafana-operator
# support specifying existing volume in grafana dataStorage

## Description
In order to reference an existing volume (e.g. for a specific blob/file-storage) it must be possible to specify the volumeName in the PVC.
The PVC will be created using the attributes in `Grafana.spec.dataStorage`, so I added a new attribute `volumeName` that will be used when creating the `PersistentVolumeClaimSpec`

## Type of change




## Checklist


## Verification steps
* Create a PersistentVolume on your Cluster and make sure, that the ReclaimPolicy is Retain (e.g. by following this guide: https://docs.microsoft.com/en-us/azure/aks/azure-files-volume#mount-file-share-as-a-persistent-volume)
* Update the CRD for Grafana on your Cluster
* Change your Grafana.yaml to reference the created volume (for an Example see `deploy/examples/persistentvolume/Grafana-existingVolume.yaml`)
* Apply the changes to the cluster and wait for the grafana-deployment POD to come up
* check that the automatically created PVC "grafana-pvc" is bound to the previously created Volume `kubectl get pvc -n your-namespace`
